app/screens/edit_profile/edit_profile.tsx

```diff
@@ -152,98 +154,148 @@ const EditProfile = ({
+            lastName: 'last_name',
+            nickname: 'nickname',
+            position: 'position',
+            username: 'username',
+            email: 'email',
+        };
+
+        const apiUpdates: Partial<UserProfile> = {};
+        Object.keys(updates).forEach((key) => {
+            const snakeCaseKey = keyMapping[key];
+            if (snakeCaseKey) {
+                (apiUpdates as any)[snakeCaseKey] = updates[key as keyof UserInfo];
+            }
+        });
+
+        return apiUpdates;
+    }, [updateUserInfo]);
+
+    const handleProfileImageUpdate = useCallback(async (serverUrlParam: string, currentUserParam: UserModel) => {
+        const localPath = changedProfilePicture.current?.localPath;
+        const profileImageRemoved = changedProfilePicture.current?.isRemoved;
+
+        if (localPath) {
+            const now = Date.now();
+            const {error: uploadError} = await uploadUserProfileImage(serverUrlParam, localPath);
+            if (uploadError) {
+                throw uploadError;
+            }
+            updateLocalUser(serverUrlParam, {last_picture_update: now});
+        } else if (profileImageRemoved) {
+            await setDefaultProfileImage(serverUrlParam, currentUserParam.id);
+        }
+    }, []);
+
+    const handleCustomAttributesUpdate = useCallback(async (
+        serverUrlParam: string,
+        currentUserParam: UserModel,
+        userInfoParam: UserInfo,
+        customAttributesSetParam: CustomAttributeSet | undefined,
+        customFieldsParam: CustomProfileFieldModel[] | undefined,
+    ) => {
+        if (!userInfoParam.customAttributes || !enableCustomAttributes) {
+            return;
+        }
+
+        const customFieldsMap = new Map<string, CustomProfileFieldModel>();
+        customFieldsParam?.forEach((field) => {
+            customFieldsMap.set(field.id, field);
+        });
+
+        const changedCustomAttributes: CustomAttributeSet = {};
+
+        Object.keys(userInfoParam.customAttributes).forEach((key) => {
+            const currentValue = (customAttributesSetParam && customAttributesSetParam[key]?.value) || '';
+            const newValue = userInfoParam.customAttributes[key]?.value || '';
+            const customAttribute = userInfoParam.customAttributes[key];
+            const customField = customFieldsMap.get(customAttribute?.id);
+
+            if (currentValue !== newValue && !isCustomFieldSamlLinked(customField)) {
+                changedCustomAttributes[key] = userInfoParam.customAttributes[key];
+            }
+        });
+
+        if (Object.keys(changedCustomAttributes).length > 0) {
+            const {error: attrError} = await updateCustomProfileAttributes(serverUrlParam, currentUserParam.id, changedCustomAttributes);
+            if (attrError) {
+                logError('Error updating custom attributes', attrError);
+                throw attrError;
+            }
+        }
+    }, [enableCustomAttributes]);
```

Comment:
Seems this can also be passed as an argument, and you can move this one also outside of the component.